packages/integration-sdk-runtime/src/logger/index.ts

```diff
@@ -295,19 +304,29 @@ export class IntegrationLogger extends EventEmitter
   }
 
   validationFailure(err: Error) {
-    const name = 'validation_failure';
+    const eventName = 'validation_failure';
     const { errorId, description } = createErrorEventDescription(
       err,
       `Error occurred while validating integration configuration.`,
     );
+    this.handleFailure({ eventName, errorId, err, description });
+  }
 
+  private handleFailure(options: {
```

Comment:
I think that if we're going to go generic and encourage switching on `eventName` in the `handleFailure` function, we should make the `eventName` a fixed set of values, such as an `enum`.

Reply:
Thanks, that's a great idea. I went with string union instead of enum since it's very specific to this interface (at the moment)